packages/turf-transform-rotate/index.d.ts

```diff
@@ -0,0 +1,13 @@
+/// <reference types="geojson" />
+import {Point} from '@turf/helpers'
+
+export type GeometryObject = GeoJSON.GeometryObject;
+export type Feature<Geom extends GeometryObject> = GeoJSON.Feature<Geom>;
+
+/**
+ * http://turfjs.org/docs/#transform-rotate
+ */
+declare function rotate(geojson: Feature<any> | GeometryObject,
```

Comment:
🥇 These Typescript definitions are not obvious, the goal with this would be to have the Input be equal to the Ouptut.

Which would equal to:
- `Geometry<Point>` => `Feature<Point>`
- `Feature<Polygon>` => `Feature<Polygon>`

Right now the current state mutates the geometry type to `any`:
- `Geometry<Point>` => `Feature<any>`
- `Feature<Polygon>` => `Feature<any>`

**Example from translate**

```typescript
export type GeometryObject = GeoJSON.GeometryObject;
export type Feature<Geom extends GeometryObject> = GeoJSON.Feature<Geom>;

/**
 * http://turfjs.org/docs/#transform-translate
 */
declare function translate<Geom extends GeometryObject>(
    geojson: Feature<Geom> | Geom,
    distance: number,
    direction: number,
    units?: string): Feature<Geom>;
```
> **Note:** I'll push a commit to update those definitions.